### DISCUSSION TITLE
关于增加一个 hugo-xmax 主题的建议

### DISCUSSION THREAD

**Discussion by chuxinyuan** [rbind/yihui#1518] [General]
hugo xmin 主题简约而不简单，已然十分稳定、轻量、快速了，是一个小而美的博客主题，我想基本上不会再有什么调整了，解决了有没有的问题。

为了满足不同用户的功能需求，建议以 hugo xmin 为基础，把各种实用的功能集成进来，比如代码块复制按钮、代码块折叠、评论等，打造一个 hugo-xmax 主题，满足 hugo xmin 用户进一步学习进阶的效果，起到示范作用。

总体上就是 TinyTeX 的套路，哈哈！

**Comment by yihui** [comment]
下限有极限，上限没有极限啊，囊括所有人需求的全家桶主题是不存在的。我当年就[已经考虑到这一点](https://yihui.org/en/2017/09/pull-requests-as-a-teaching-tool/)，所以已经在这里给了一些功能实现的示范：https://github.com/yihui/hugo-xmin/pulls 帮人帮不到底，送佛送不到西；师傅领进门，修行靠个人了。最近我在写一系列常见网页功能的几行代码“极简实现”，同样是这个目的。

**Reply by chuxinyuan** [reply]
大部分人的角色可能是用户而不是开发人员，一般用户估计不会有太多心思和时间打磨一个主题。xmax 不用取极限（就好像不用把所有的 R包都装上一样），只要把常用的纳入即可，至于用不用，用户只需要在 config.yaml 里设置 yes 或 no 即可。剩下的交给其他开发人员，类似 Linux 的开发过程，他们如果有好的功能点和创意，可以放进来。同时，xmax 演进的过程，可以为 blogdown 书的提供更多素材。

以上个人愚见。

**Reply by yihui** [reply]
我暂时没有时间弄这个，也不是很有兴趣，你可以考虑做这件事啊。我明白你的意思，只是我现在有点倾向于 Linux 那种从源码编译起的哲学，就是我把最基础的代码丢给用户并讲明白，让有缘人自己去基础层面组合组装。

**Reply by chuxinyuan** [reply]
> 我暂时没有时间弄这个，也不是很有兴趣，你可以考虑做这件事啊。

恩，我认真考虑下。

> 我现在有点倾向于 Linux 那种从源码编译起的哲学，就是我把最基础的代码丢给用户并讲明白，让有缘人自己去基础层面组合组装。

自己动手搞定可玩性比较高，而且一旦自己搞定了很有成就感，我的下一个系统或许会是 Arch Linux。如果不是为了生计，我真想宅家里把 Mnix 或者 Linus 最初的那个 Linux 版本复现一遍。
